src/modules/cpu/host_tensor_augmentations.hpp

```diff
@@ -5265,4 +5265,549 @@ RppStatus color_cast_i8_i8_host_tensor(Rpp8s *srcPtr,
+        srcPtrImage2 = srcPtr + (permutedIndices2[batchCount] * srcDescPtr->strides.nStride);
+        srcPtrImage3 = srcPtr + (permutedIndices3[batchCount] * srcDescPtr->strides.nStride);
+        srcPtrImage4 = srcPtr + (permutedIndices4[batchCount] * srcDescPtr->strides.nStride);
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength1 = roiPtr1->xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u bufferLength2 = roiPtr2->xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u bufferLength3 = roiPtr3->xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u bufferLength4 = roiPtr4->xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp8u *srcPtrChannel1, *srcPtrChannel2, *srcPtrChannel3, *srcPtrChannel4, *dstPtrChannel;
+        srcPtrChannel1 = srcPtrImage1 + (roiPtr1->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr1->xywhROI.xy.x * layoutParams.bufferMultiplier);
+        srcPtrChannel2 = srcPtrImage2 + (roiPtr2->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr2->xywhROI.xy.x * layoutParams.bufferMultiplier);
+        srcPtrChannel3 = srcPtrImage3 + (roiPtr3->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr3->xywhROI.xy.x * layoutParams.bufferMultiplier);
+        srcPtrChannel4 = srcPtrImage4 + (roiPtr4->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr4->xywhROI.xy.x * layoutParams.bufferMultiplier);
+
+        dstPtrChannel = dstPtrImage;
+        // ricap with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp32u alignedLength1 = bufferLength1 & ~47;
+            Rpp32u alignedLength2 = bufferLength2 & ~47;
+            Rpp32u alignedLength3 = bufferLength3 & ~47;
+            Rpp32u alignedLength4 = bufferLength4 & ~47;
+
+            Rpp8u *srcPtrRow1, *srcPtrRow2, *srcPtrRow3, *srcPtrRow4, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            srcPtrRow1 = srcPtrChannel1;
+            srcPtrRow2 = srcPtrChannel2;
+            srcPtrRow3 = srcPtrChannel3;
+            srcPtrRow4 = srcPtrChannel4;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int i = 0; i < roiPtr1->xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTemp1, *srcPtrTemp2, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                srcPtrTemp1 = srcPtrRow1;
+                srcPtrTemp2 = srcPtrRow2;
+                
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+
+                int vectorLoopCount1 = 0;
+                for (; vectorLoopCount1 < alignedLength1; vectorLoopCount1+=48)
+                {
+                    __m128 p[3];
```

Comment:
Change this to __m128i px[3]; They should all be __m128i registers. And all other dependent changes needed, including the rpp_load48_u8pkd3_to_u8pln3() function that should receive px;

Reply:
Done